apps/maestro/src/lib/utils/chains.ts

```diff
@@ -42,15 +41,22 @@ export const filterEligibleChains = (
   const isAllChainsWhitelisted = whitelistedChains[0] === "all";
 
   return destinationChains.filter((chain) => {
-    // Always filter out current chain
-    if (chain.chain_id === currentChainId) return false;
+    const isOriginChainSameAsDestinationChain =
+      chain.chain_id === currentChainId;
+    const areDifferentChainTypes = chain.chain_type !== currentChain.chain_type;
+    const isThisDestinationChainWhitelisted =
+      isAllChainsWhitelisted || whitelistedChains.includes(chain.id);
+    const isOriginChainWhitelisted =
+      isAllChainsWhitelisted || whitelistedChains.includes(currentChain?.id);
 
-    // For EVM chains, check whitelist
-    if (isCurrentChainVM && chain.chain_type === "evm") {
-      return isAllChainsWhitelisted || whitelistedChains.includes(chain.id);
-    }
-
-    // Non-EVM chains are always included
+    // Always filter out current chain
+    if (isOriginChainSameAsDestinationChain) return false;
+    // Whitelisted chains are always included
+    if (isThisDestinationChainWhitelisted || isOriginChainWhitelisted)
+      return true;
```

Comment:
so this returns true for `sui -> any evm chains` since sui is the `isOriginChainWhitelisted`?